### ISSUE TITLE
Serverside redux+socket example

### ISSUE THREAD

**Issue by underscoredam** [tomatau/breko-hub#259]
Hi,

I want to build a client <-> server socket connection in one of my routes. I'm not exactly sure how to do that (I couldn't `import socket from 'composition/socket.js'` in a route due to `TypeError: Cannot read property 'protocol' of undefined`). Can I get a sample code for event listeners for both server sided API route and client side `Component`? Thanks.

**Comment by tomatau** [comment]
Hey! Thanks for asking! Sorry I won't be able to contribute much time at the moment as I'm overwhelmed AFK but maybe this can help as it's already integrated

https://github.com/rstuven/redux-via-socket.io

You can use redux for socket communication by adding `meta { broadcast: true }` or the other options specified here https://github.com/rstuven/redux-via#meta-options

**Comment by underscoredam** [comment]
Okay, I was able to figure out how to do this with your help. I am thinking about writing an example and comment to this issue later this week. I noticed two things while I was working on this:

1. There is no way to dispatch from the server since the `socketsStore` is never accessible to routes. 
2. On the same file, `sockets.js` is importing createStore from `src/app/compositions`. I thought that `src/app` was for client side application. I do not want the same store for client and server (even though I am fine with this). Is this an intended feature?

Thank you! Your repository is super organized and I'm glad that I chose to use this! One more thing I'd like to comment is the git feature to prevent pushing while the tests fail. While I like this feature, I am workin on a huge feature for my own project and I need to constantly push my commits. I can override the githooks but I was in doubt whether we should enforce this rule for branches other than the master (or the upstream branch).
